packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -249,6 +261,79 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         return db.recoveryKeyExists(uid);
       },
     },
+    {
+      method: 'GET',
+      path: '/recoveryKey/hint',
+      options: {
+        ...RECOVERY_KEY_DOCS.RECOVERYKEY_HINT_GET,
+        // No auth needed to fetch the recovery key hint
+        auth: false,
+        validate: {
+          query: {
+            uid: isA.string().min(32).max(32).regex(HEX_STRING),
+          },
+        },
+      },
+      handler: async function (request) {
+        log.begin('getRecoveryKeyHint', request);
+
+        const { uid } = request.query;
+
+        const account = await db.account(uid);
+        const email = account.primaryEmail.email;
```

Comment:
Q: Do we know that `account` will return a record and not `null`/`undefined` with the query string `uid` and cause potential issues here when trying to access `.primaryEmail`?

Reply:
`getRecoveryKeyHint` should only be called if we've already verified that the user has an account, but it makes sense to add a safety net.